jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -709,6 +710,79 @@ ecma_concat_ecma_strings (ecma_string_t *string1_p, /**< first ecma-string */
 } /* ecma_concat_ecma_strings */
 
 /**
+ * Format ecma-strings.
+ *
+ * @return a newly created ecma-string
+ */
+ecma_string_t *
+ecma_format_ecma_strings (const char * format, /**< format string */
+                          ...) /**< ecma-strings */
+{
+  JERRY_ASSERT (format != NULL);
+
+  va_list args;
+
+  va_start (args, format);
+
+  ecma_string_t *string_p = ecma_get_magic_string (LIT_MAGIC_STRING__EMPTY);
+  ecma_string_t *string1_p;
+  ecma_string_t *string2_p;
+
+  const char *start_p = format;
+  const char *end_p = format;
+
+  while (*end_p)
+  {
+    if (*end_p == '%')
+    {
+      if (likely (*(end_p + 1) != '%'))
+      {
+        string1_p = string_p;
+        string2_p = ecma_new_ecma_string_from_utf8 ((const lit_utf8_byte_t *) start_p,
+                                                    (lit_utf8_size_t) (end_p - start_p));
+        string_p = ecma_concat_ecma_strings (string1_p, string2_p);
+        ecma_deref_ecma_string (string1_p);
+        ecma_deref_ecma_string (string2_p);
+
+        string1_p = string_p;
+        string2_p = va_arg (args, ecma_string_t *);
+        string_p = ecma_concat_ecma_strings (string1_p, string2_p);
+        ecma_deref_ecma_string (string1_p);
+      }
+      else
+      {
+        end_p++;
+
+        string1_p = string_p;
+        string2_p = ecma_new_ecma_string_from_utf8 ((const lit_utf8_byte_t *) start_p,
+                                                    (lit_utf8_size_t) (end_p - start_p));
+        string_p = ecma_concat_ecma_strings (string1_p, string2_p);
+        ecma_deref_ecma_string (string1_p);
+        ecma_deref_ecma_string (string2_p);
+      }
```

Comment:
Do we need to support double `%` sign? This function is not a general printf, we probably don't need to overcomplicate things.